### PR TITLE
Remove `buf config` and move commands to `mod` and `beta` respectively

### PR DESCRIPTION
**Pull request by doriable**
After some discussion, we've decided to remove `buf config` and consolidate most commands under `buf mod`.
The only exception is the `migrate-v1beta1` command, which has been moved to `buf beta`.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -21,6 +21,10 @@
   details.
 - Context modifier no longer overrides an existing token on the context. This allows `buf registry login`
   to properly check the user provided token without the token being overriden by the CLI interceptor.
+- Removed the `buf config init` command in favor of `buf mod init`.
+- Removed the `buf config ls-breaking-rules` command in favor of `buf mod ls-breaking-rules`.
+- Removed the `buf config ls-lint-rules` command in favor of `buf mod ls-lint-rules`.
+- Removed the `buf config migrate-v1beta1` command in favor of `buf beta config migrate-v1beta1`.
```
Comment (amckinney) [inline CHANGELOG.md:27]: I think we might have just wanted `buf beta migrate-v1beta1` (cc @bufdev). I could be mistaken.

File: private/buf/cmd/buf/command/mod/modlslintrules/modlslintrules.go [hunk 2]
```diff
@@ -67,10 +67,10 @@ func newFlags() *flags {
...
 
 func (f *flags) Bind(flagSet *pflag.FlagSet) {
-	configinternal.BindLSRulesAll(flagSet, &f.All, allFlagName)
-	configinternal.BindLSRulesConfig(flagSet, &f.Config, configFlagName, allFlagName, versionFlagName)
-	configinternal.BindLSRulesFormat(flagSet, &f.Format, formatFlagName)
-	configinternal.BindLSRulesVersion(flagSet, &f.Version, versionFlagName, allFlagName)
+	internal.BindLSRulesAll(flagSet, &f.All, allFlagName)
+	internal.BindLSRulesConfig(flagSet, &f.Config, configFlagName, allFlagName, versionFlagName)
+	internal.BindLSRulesFormat(flagSet, &f.Format, formatFlagName)
+	internal.BindLSRulesVersion(flagSet, &f.Version, versionFlagName, allFlagName)
```
Comment (amckinney) [inline private/buf/cmd/buf/command/mod/modlslintrules/modlslintrules.go:73]: These should probably allo be `modinternal` (based on the previous import).

File: CHANGELOG.md (hunk 1 above)
Comment (doriable) [reply CHANGELOG.md:27]: I am onboard with that, I wasn't sure how granular we wanted the `beta` namespace ^^"

File: CHANGELOG.md [hunk 3]
```diff
@@ -21,6 +21,10 @@
   details.
 - Context modifier no longer overrides an existing token on the context. This allows `buf registry login`
   to properly check the user provided token without the token being overriden by the CLI interceptor.
+- Removed the `buf config init` command in favor of `buf mod init`.
+- Removed the `buf config ls-breaking-rules` command in favor of `buf mod ls-breaking-rules`.
+- Removed the `buf config ls-lint-rules` command in favor of `buf mod ls-lint-rules`.
+- Removed the `buf config migrate-v1beta1` command in favor of `buf beta migrate-v1beta1`.
```
Comment (amckinney) [inline CHANGELOG.md:27]: This should have been done in the other PR, but could you add a note for adding the `buf beta decode` command in the `CHANGELOG.md`? For reference, see https://github.com/bufbuild/buf/pull/925.

**Review by amckinney — APPROVED** [review]

**Pull request merged**